fork-off/src/main.rs

```diff
@@ -12,168 +19,336 @@ pub struct Config {
     #[clap(long, default_value = "http://127.0.0.1:9933")]
     pub http_rpc_endpoint: String,
 
-    /// path to write the initial chainspec of the fork
-    /// as generated with the `bootstrap-chain` command
-    #[clap(long, default_value = "../docker/data/chainspec.json")]
-    pub fork_spec_path: String,
+    /// path of the initial chainspec (generated with the `bootstrap-chain` command)
+    #[clap(long, default_value = "./initial_chainspec.json")]
+    pub initial_spec_path: String,
+
+    /// where to write the snapshot of the state
+    #[clap(long, default_value = "./snapshot.json")]
+    pub snapshot_path: String,
+
+    /// where to write the forked genesis chainspec
+    #[clap(long, default_value = "./chainspec_from_snapshot.json")]
+    pub combined_spec_path: String,
 
     /// where to write the forked genesis chainspec
-    #[clap(long, default_value = "../docker/data/chainspec.fork.json")]
-    pub write_to_path: String,
+    #[clap(long)]
+    pub use_snapshot_file: bool,
+
+    /// how many parallel processes to download values -- note that large values might result in bans because
+    /// of rate-limiting mechanisms
+    #[clap(long, default_value_t = 5)]
+    pub num_workers: u32,
 
     /// which modules to set in forked spec
     #[clap(
         long,
         multiple_occurrences = true,
         takes_value = true,
         value_delimiter = ',',
-        default_value = "Aura,Aleph"
+        default_value = "Aura,Aleph,Sudo,Staking,Session,Elections"
     )]
-    pub prefixes: Vec<String>,
+    pub pallets: Vec<String>,
```

Comment:
Maybe a rename of the argument is in order:
`--keep-state-for-pallets`,  `--idempotent-pallets` (to get fancy) or something to that effect?

Reply:
Renamed by `pallets_keep_state`